libs/dh/market-participant/data-access-api/src/lib/dh-market-participant-edit-actor-data-access-api.store.ts

```diff
@@ -148,16 +134,88 @@ export class DhMarketParticipantEditActorDataAccessApiStore extends ComponentSto
     )
   );
 
+  readonly loadInitialData = this.effect(
+    (routeParams$: Observable<{ organizationId: string; actorId: string }>) =>
+      routeParams$.pipe(
+        switchMap((x) =>
+          forkJoin({
+            gridAreas: this.getGridAreas(),
+            actorAndContacts: this.getActorAndContacts(x),
+          }).pipe(
+            tap((x) =>
+              this.patchState({
+                gridAreaChanges: x.gridAreas.filter((y) =>
+                  x.actorAndContacts.gridAreas.includes(y.id)
+                ),
+              })
+            )
+          )
+        )
+      )
+  );
+
+  readonly getGridAreas = () =>
+    of('').pipe(
+      tap(() => this.patchState({ isLoadingGridAreas: true })),
+      switchMap(() =>
+        this.gridAreaHttpClient
```

Comment:
suggestion: if `isLoadingGridAreas` and `isLoadingActor` are merged to a single property, we can return the call to the endpoint directly and set `isLoading` state in the caller method.

Reply:
fixed @dzhavat